File: testdata/apps/php-worker/worker.php
<?php
// Long-running PHP worker with SIGTERM handling via pcntl. Emits a
// heartbeat line each second and exits 0 on SIGTERM / SIGINT so the
// supervisor sees a clean graceful stop — mirrors python-worker and
// ruby-worker so any runtime-specific regression stands out.

declare(strict_types=1);

pcntl_async_signals(true);
pcntl_signal(SIGTERM, static function (): void {
    fwrite(STDOUT, "php-worker received SIGTERM, exiting\n");
    exit(0);
});
pcntl_signal(SIGINT, static function (): void {
    fwrite(STDOUT, "php-worker received SIGINT, exiting\n");
    exit(0);
});

fwrite(STDOUT, sprintf("php-worker pid=%d\n", getmypid()));
$tick = 0;
while (true) {
    fwrite(STDOUT, sprintf("php-worker tick=%d\n", $tick));
    $tick++;
    sleep(1);
}

File: testdata/apps/python-worker/worker.py
#!/usr/bin/env python3
"""Long-running worker. Emits a heartbeat line each second and exits 0
on SIGTERM, so the supervisor sees a clean graceful stop."""

import os
import signal
import sys
import time

running = True


def _shutdown(sig, _frame):
    global running
    print(f"python-worker received signal {sig}, exiting", flush=True)
    running = False


signal.signal(signal.SIGTERM, _shutdown)
signal.signal(signal.SIGINT, _shutdown)

print(f"python-worker pid={os.getpid()}", flush=True)
tick = 0
while running:
    print(f"python-worker tick={tick}", flush=True)
    tick += 1
    time.sleep(1)

sys.exit(0)
